MNE/disp3D/3DObjects/brain/brainrtconnectivitydatatreeitem.cpp

```diff
@@ -144,14 +142,161 @@ bool BrainRTConnectivityDataTreeItem::init(const MNEForwardSolution& tForwardSol
+        qDebug() << "BrainRTConnectivityDataTreeItem::updateData - BrainRTConnectivityDataTreeItem has not been initialized yet!";
         return false;
     }
 
+    //Add data which is held by this BrainRTConnectivityDataTreeItem
+    QVariant data;
+
+    data.setValue(pNetworkData);
+    this->setData(data, Data3DTreeModelItemRoles::NetworkData);
+
+    MatrixXd matDist = pNetworkData->getConnectivityMatrix();
+    data.setValue(matDist);
+    this->setData(data, Data3DTreeModelItemRoles::NetworkDataMatrix);
+
+    //Add surface meta information as item children
+    QList<QStandardItem*> list;
+
+    QVector3D vecEdgeTrehshold(0,0,0);
+    MetaTreeItem* pItemNetworkThreshold = new MetaTreeItem(MetaTreeItemTypes::NetworkThreshold, "0.0,0.0,0.0");
+    connect(pItemNetworkThreshold, &MetaTreeItem::networkThresholdChanged,
+            this, &BrainRTConnectivityDataTreeItem::onNetworkThresholdChanged);
+    list << pItemNetworkThreshold;
+    list << new QStandardItem(pItemNetworkThreshold->toolTip());
+    this->appendRow(list);
+    data.setValue(vecEdgeTrehshold);
+    pItemNetworkThreshold->setData(data, MetaTreeItemRoles::NetworkThreshold);
+
+    list.clear();
+    MetaTreeItem* pItemNetworkMatrix = new MetaTreeItem(MetaTreeItemTypes::NetworkMatrix, "Show network matrix");
+    list << pItemNetworkMatrix;
+    list << new QStandardItem(pItemNetworkMatrix->toolTip());
+    this->appendRow(list);
+
+    //Plot network
+    plotNetwork(pNetworkData, vecEdgeTrehshold);
+
     return true;
 }
 
+
+//*************************************************************************************************************
+
+void BrainRTConnectivityDataTreeItem::onCheckStateChanged(const Qt::CheckState& checkState)
+{
+    this->setVisible(checkState == Qt::Unchecked ? false : true);
+}
+
+
+//*************************************************************************************************************
+
+void BrainRTConnectivityDataTreeItem::setVisible(bool state)
+{
+    for(int i = 0; i < m_lNodes.size(); ++i) {
+        m_lNodes.at(i)->setParent(state ? m_pRenderable3DEntity : Q_NULLPTR);
+    }
+
+    m_pRenderable3DEntity->setParent(state ? m_pParentEntity : Q_NULLPTR);
+}
+
+
+//*************************************************************************************************************
+
+void BrainRTConnectivityDataTreeItem::onNetworkThresholdChanged(const QVector3D& vecThresholds)
+{
+    qDebug()<<"";
+    Network::SPtr pNetwork = this->data(Data3DTreeModelItemRoles::NetworkData).value<Network::SPtr>();
+
+    plotNetwork(pNetwork, vecThresholds);
+}
+
+
+//*************************************************************************************************************
+
+void BrainRTConnectivityDataTreeItem::plotNetwork(QSharedPointer<CONNECTIVITYLIB::Network> pNetworkData, const QVector3D& vecThreshold)
+{
+    //Prepare network visualization
+    QMatrix4x4 m;
+    Qt3DCore::QTransform* transform =  new Qt3DCore::QTransform();
```

Comment:
QEntity does not accpet smart/shared pointers. Note: If the Qt3DCore::QComponent has no parent, the Qt3DCore::QEntity will set itself as its parent thereby taking ownership of the component. This call should therefore be safe. What do you think?